### ISSUE TITLE
Error in './spawn': double free or corruption (!prev)

### ISSUE THREAD

**Issue by borlum** [NREL/Spawn#44]
So, I need to compile spawn in order to have it run on ancient RHEL7 - as the releases you distribute depend on GLIBC > 2.25, and RHEL is only GLIBC 2.17. Compilation of tag v0.2.1 works, but I run into the following runtime error:

`
*** Error in './spawn': double free or corruption (!prev): 0x00000000035988c0 ***
`

And here I get a little bit stuck. You can see the full dump from the error below.

Kudos on a cool project - really hoping I can get this to work!

----

$ ./spawn

*** Error in './spawn': double free or corruption (!prev): 0x00000000035988c0 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x81299)[0x2aaab019c299]
/lib64/libc.so.6(__cxa_finalize+0x9a)[0x2aaab015505a]
/home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/bin/../lib/libenergyplusapi.so.9.5.0(+0xaa8fa3)[0x2aaaabcd5fa3]
======= Memory map: ========
00400000-00598000 r--p 00000000 00:2c 16377000427503977893               /home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/bin/spawn-0.2.1-5d127d5774
00598000-02c7c000 r-xp 00198000 00:2c 16377000427503977893               /home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/bin/spawn-0.2.1-5d127d5774
02c7c000-034e1000 r--p 0287c000 00:2c 16377000427503977893               /home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/bin/spawn-0.2.1-5d127d5774
034e1000-03521000 r--p 030e0000 00:2c 16377000427503977893               /home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/bin/spawn-0.2.1-5d127d5774
03521000-03527000 rw-p 03120000 00:2c 16377000427503977893               /home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/bin/spawn-0.2.1-5d127d5774
03527000-035b9000 rw-p 00000000 00:00 0                                  [heap]
2aaaaaaab000-2aaaaaacd000 r-xp 00000000 08:13 11153053                   /usr/lib64/ld-2.17.so
2aaaaaacd000-2aaaaaacf000 r-xp 00000000 00:00 0                          [vdso]
2aaaaaacf000-2aaaaaad2000 rw-p 00000000 00:00 0 
2aaaaaaf4000-2aaaaab01000 rw-p 00000000 00:00 0 
2aaaaaccc000-2aaaaaccd000 r--p 00021000 08:13 11153053                   /usr/lib64/ld-2.17.so
2aaaaaccd000-2aaaaacce000 rw-p 00022000 08:13 11153053                   /usr/lib64/ld-2.17.so
2aaaaacce000-2aaaaaccf000 rw-p 00000000 00:00 0 
2aaaaaccf000-2aaaaacd1000 r-xp 00000000 08:13 11154507                   /usr/lib64/libdlfaker.so
2aaaaacd1000-2aaaaaed0000 ---p 00002000 08:13 11154507                   /usr/lib64/libdlfaker.so
2aaaaaed0000-2aaaaaed1000 rw-p 00001000 08:13 11154507                   /usr/lib64/libdlfaker.so
2aaaaaed1000-2aaaaaff4000 r-xp 00000000 08:13 11157580                   /usr/lib64/libvglfaker.so
2aaaaaff4000-2aaaab1f3000 ---p 00123000 08:13 11157580                   /usr/lib64/libvglfaker.so
2aaaab1f3000-2aaaab1f9000 rw-p 00122000 08:13 11157580                   /usr/lib64/libvglfaker.so
2aaaab1f9000-2aaaab22d000 rw-p 00000000 00:00 0 
2aaaab22d000-2aaaaed7e000 r-xp 00000000 00:2c 15396952036074685557       /home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/lib/libenergyplusapi.so.9.5.0
2aaaaed7e000-2aaaaef7d000 ---p 03b51000 00:2c 15396952036074685557       /home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/lib/libenergyplusapi.so.9.5.0
2aaaaef7d000-2aaaaefcd000 r--p 03b50000 00:2c 15396952036074685557       /home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/lib/libenergyplusapi.so.9.5.0
2aaaaefcd000-2aaaaeff2000 rw-p 03ba0000 00:2c 15396952036074685557       /home/u101337/Spawn-light-0.2.1-5d127d5774-Linux/lib/libenergyplusapi.so.9.5.0
2aaaaeff2000-2aaaaf009000 rw-p 00000000 00:00 0 
2aaaaf009000-2aaaaf020000 r-xp 00000000 08:13 11156610                   /usr/lib64/libpthread-2.17.so
2aaaaf020000-2aaaaf21f000 ---p 00017000 08:13 11156610                   /usr/lib64/libpthread-2.17.so
2aaaaf21f000-2aaaaf220000 r--p 00016000 08:13 11156610                   /usr/lib64/libpthread-2.17.so
2aaaaf220000-2aaaaf221000 rw-p 00017000 08:13 11156610                   /usr/lib64/libpthread-2.17.so
2aaaaf221000-2aaaaf225000 rw-p 00000000 00:00 0 
2aaaaf225000-2aaaaf227000 r-xp 00000000 08:13 11154276                   /usr/lib64/libdl-2.17.so
2aaaaf227000-2aaaaf427000 ---p 00002000 08:13 11154276                   /usr/lib64/libdl-2.17.so
2aaaaf427000-2aaaaf428000 r--p 00002000 08:13 11154276                   /usr/lib64/libdl-2.17.so
2aaaaf428000-2aaaaf429000 rw-p 00003000 08:13 11154276                   /usr/lib64/libdl-2.17.so
2aaaaf429000-2aaaaf5e6000 r-xp 00000000 00:2c 5352867457449803337        /beegfs/apps/gcc/10.2.0/lib64/libstdc++.so.6.0.28
2aaaaf5e6000-2aaaaf7e5000 ---p 001bd000 00:2c 5352867457449803337        /beegfs/apps/gcc/10.2.0/lib64/libstdc++.so.6.0.28
2aaaaf7e5000-2aaaaf7f0000 r--p 001bc000 00:2c 5352867457449803337        /beegfs/apps/gcc/10.2.0/lib64/libstdc++.so.6.0.28
2aaaaf7f0000-2aaaaf7f3000 rw-p 001c7000 00:2c 5352867457449803337        /beegfs/apps/gcc/10.2.0/lib64/libstdc++.so.6.0.28
2aaaaf7f3000-2aaaaf7f6000 rw-p 00000000 00:00 0 
2aaaaf7f6000-2aaaaf7f8000 r-xp 00000000 08:13 11155096                   /usr/lib64/libutil-2.17.so
2aaaaf7f8000-2aaaaf9f7000 ---p 00002000 08:13 11155096                   /usr/lib64/libutil-2.17.so
2aaaaf9f7000-2aaaaf9f8000 r--p 00001000 08:13 11155096                   /usr/lib64/libutil-2.17.so
2aaaaf9f8000-2aaaaf9f9000 rw-p 00002000 08:13 11155096                   /usr/lib64/libutil-2.17.so
2aaaaf9f9000-2aaaafa00000 r-xp 00000000 08:13 11154936                   /usr/lib64/librt-2.17.so
2aaaafa00000-2aaaafbff000 ---p 00007000 08:13 11154936                   /usr/lib64/librt-2.17.so
2aaaafbff000-2aaaafc00000 r--p 00006000 08:13 11154936                   /usr/lib64/librt-2.17.so
2aaaafc00000-2aaaafc01000 rw-p 00007000 08:13 11154936                   /usr/lib64/librt-2.17.so
2aaaafc01000-2aaaafd02000 r-xp 00000000 08:13 11154832                   /usr/lib64/libm-2.17.so
2aaaafd02000-2aaaaff01000 ---p 00101000 08:13 11154832                   /usr/lib64/libm-2.17.so
2aaaaff01000-2aaaaff02000 r--p 00100000 08:13 11154832                   /usr/lib64/libm-2.17.so
2aaaaff02000-2aaaaff03000 rw-p 00101000 08:13 11154832                   /usr/lib64/libm-2.17.so
2aaaaff03000-2aaaaff1a000 r-xp 00000000 00:2c 16647705671592783968       /beegfs/apps/gcc/10.2.0/lib64/libgcc_s.so.1
2aaaaff1a000-2aaab0119000 ---p 00017000 00:2c 16647705671592783968       /beegfs/apps/gcc/10.2.0/lib64/libgcc_s.so.1
2aaab0119000-2aaab011a000 r--p 00016000 00:2c 16647705671592783968       /beegfs/apps/gcc/10.2.0/lib64/libgcc_s.so.1
2aaab011a000-2aaab011b000 rw-p 00017000 00:2c 16647705671592783968       /beegfs/apps/gcc/10.2.0/lib64/libgcc_s.so.1
2aaab011b000-2aaab02df000 r-xp 00000000 08:13 11154214                   /usr/lib64/libc-2.17.so
2aaab02df000-2aaab04de000 ---p 001c4000 08:13 11154214                   /usr/lib64/libc-2.17.so
2aaab04de000-2aaab04e2000 r--p 001c3000 08:13 11154214                   /usr/lib64/libc-2.17.so
2aaab04e2000-2aaab04e4000 rw-p 001c7000 08:13 11154214                   /usr/lib64/libc-2.17.so
2aaab04e4000-2aaab04e9000 rw-p 00000000 00:00 0 
2aaab04e9000-2aaab055a000 r-xp 00000000 08:13 11154066                   /usr/lib64/libGL.so.1.7.0
2aaab055a000-2aaab0759000 ---p 00071000 08:13 11154066                   /usr/lib64/libGL.so.1.7.0
2aaab0759000-2aaab0773000 r--p 00070000 08:13 11154066                   /usr/lib64/libGL.so.1.7.0
2aaab0773000-2aaab0774000 rw-p 0008a000 08:13 11154066                   /usr/lib64/libGL.so.1.7.0
2aaab0774000-2aaab0775000 rw-p 00000000 00:00 0 
2aaab0775000-2aaab0779000 r-xp 00000000 08:13 11154150                   /usr/lib64/libXv.so.1.0.0
2aaab0779000-2aaab0978000 ---p 00004000 08:13 11154150                   /usr/lib64/libXv.so.1.0.0
2aaab0978000-2aaab0979000 r--p 00003000 08:13 11154150                   /usr/lib64/libXv.so.1.0.0
2aaab0979000-2aaab097a000 rw-p 00004000 08:13 11154150                   /usr/lib64/libXv.so.1.0.0
2aaab097a000-2aaab0ab2000 r-xp 00000000 08:13 11154124                   /usr/lib64/libX11.so.6.3.0
2aaab0ab2000-2aaab0cb2000 ---p 00138000 08:13 11154124                   /usr/lib64/libX11.so.6.3.0
2aaab0cb2000-2aaab0cb3000 r--p 00138000 08:13 11154124                   /usr/lib64/libX11.so.6.3.0
2aaab0cb3000-2aaab0cb8000 rw-p 00139000 08:13 11154124                   /usr/lib64/libX11.so.6.3.0
2aaab0cb8000-2aaab0cc9000 r-xp 00000000 08:13 11154133                   /usr/lib64/libXext.so.6.4.0
2aaab0cc9000-2aaab0ec8000 ---p 00011000 08:13 11154133                   /usr/lib64/libXext.so.6.4.0
2aaab0ec8000-2aaab0ec9000 r--p 00010000 08:13 11154133                   /usr/lib64/libXext.so.6.4.0
2aaab0ec9000-2aaab0eca000 rw-p 00011000 08:13 11154133                   /usr/lib64/libXext.so.6.4.0
2aaab0eca000-2aaab0eda000 r-xp 00000000 08:13 11154072                   /usr/lib64/libGLX.so.0.0.0
2aaab0eda000-2aaab10da000 ---p 00010000 08:13 11154072                   /usr/lib64/libGLX.so.0.0.0
2aaab10da000-2aaab10db000 r--p 00010000 08:13 11154072                   /usr/lib64/libGLX.so.0.0.0
2aaab10db000-2aaab10dc000 rw-p 00011000 08:13 11154072                   /usr/lib64/libGLX.so.0.0.0
2aaab10dc000-2aaab10fc000 rw-p 00000000 00:00 0 
2aaab10fc000-2aaab116a000 r-xp 00000000 08:13 11154075                   /usr/lib64/libGLdispatch.so.0.0.0
2aaab116a000-2aaab1369000 ---p 0006e000 08:13 11154075                   /usr/lib64/libGLdispatch.so.0.0.0
2aaab1369000-2aaab1391000 r--p 0006d000 08:13 11154075                   /usr/lib64/libGLdispatch.so.0.0.0
2aaab1391000-2aaab1392000 rw-p 00095000 08:13 11154075                   /usr/lib64/libGLdispatch.so.0.0.0
2aaab1392000-2aaab13b2000 rw-p 00000000 00:00 0 
2aaab13b2000-2aaab13d9000 r-xp 00000000 08:13 11155289                   /usr/lib64/libxcb.so.1.1.0
2aaab13d9000-2aaab15d8000 ---p 00027000 08:13 11155289                   /usr/lib64/libxcb.so.1.1.0
2aaab15d8000-2aaab15d9000 r--p 00026000 08:13 11155289                   /usr/lib64/libxcb.so.1.1.0
2aaab15d9000-2aaab15da000 rw-p 00027000 08:13 11155289                   /usr/lib64/libxcb.so.1.1.0
2aaab15da000-2aaab15dc000 r-xp 00000000 08:13 11154126                   /usr/lib64/libXau.so.6.0.0
2aaab15dc000-2aaab17dc000 ---p 00002000 08:13 11154126                   /usr/lib64/libXau.so.6.0.0
2aaab17dc000-2aaab17dd000 r--p 00002000 08:13 11154126                   /usr/lib64/libXau.so.6.0.0
2aaab17dd000-2aaab17de000 rw-p 00003000 08:13 11154126                   /usr/lib64/libXau.so.6.0.0
2aaab4000000-2aaab4021000 rw-p 00000000 00:00 0 
2aaab4021000-2aaab8000000 ---p 00000000 00:00 0 
7ffffffdc000-7ffffffff000 rw-p 00000000 00:00 0                          [stack]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Aborted (core dumped)

**Comment by borlum** [comment]
I just now tried compiling it on Ubuntu 18.04 as per your instructions - and the almost same runtime error appears:

$ ./spawn
free(): double free detected in tcache 2
Aborted

It appears to be an issue with the libenergyplusapi.so.9.5.0 compiled. If I download the "light" version from your available releases, I get the same runtime error when dynamically linking with the libenergyplusapi.so.9.5 that I compiled.

How do I compile the non-"light" version where libenergyplusapi is not a dependency?
